sound/soc/soc-topology.c

```diff
@@ -2093,6 +2106,10 @@ static int soc_tplg_link_config(struct soc_tplg *tplg,
 		dev_err(tplg->dev, "ASoC: physical link loading failed\n");
 		return ret;
 	}
+	link->dobj.index = tplg->index;
+	link->dobj.ops = tplg->ops;
+	link->dobj.type = SND_SOC_DOBJ_NONE;
```

Comment:
is this a hack or really a type we want to use?

Reply:
@bardliao I think we are contradicting ourselves by calling it type SND_SOC_DOBJ_NONE and then adding it to the dobj_list. 

I think the better option is to use SND_SOC_DOBJ_DAI_LINK type and re-use the remove_link to check for both passes, SOC_TPLG_PASS_PCM_DAI and SOC_TPLG_PASS_LINK.

Reply:
@ranj063 There is a problem if we reuse SND_SOC_DOBJ_DAI_LINK. [soc_remove_dai_links](https://elixir.bootlin.com/linux/latest/source/sound/soc/soc-core.c#L1018) will check if the link is SND_SOC_DOBJ_DAI_LINK type. I think soc-core think all SND_SOC_DOBJ_DAI_LINK type is created by topology. So we will see "Topology forgot to remove link <link name>" warning if we reuse  SND_SOC_DOBJ_DAI_LINK type.

Reply:
@bardliao do BE links that are defined in the machine driver get added to the card's dai_link_list?

Then maybe we need to modify the check to make sure it is a BE link before we generate the warning.